// @workspace/api-zod/src/generated/api.ts
/**
 * Generated by orval v8.5.3 🍺
 * Do not edit manually.
 * Api
 * API specification
 * OpenAPI spec version: 0.1.0
 */
import * as zod from "zod";

/**
 * Returns server health status
 * @summary Health check
 */
export const HealthCheckResponse = zod.object({
  status: zod.string(),
});

/**
 * Authenticate with email and password, returns a 24h JWT
 * @summary Admin login
 */
export const AdminLoginBody = zod.object({
  email: zod.string().email(),
  password: zod.string(),
});

export const AdminLoginResponse = zod.object({
  token: zod.string(),
  admin: zod.object({
    id: zod.number(),
    email: zod.string(),
    name: zod.string(),
  }),
});

/**
 * Aggregated metrics for the admin dashboard home page
 * @summary Dashboard statistics
 */
export const GetDashboardStatsResponse = zod.object({
  totalLeads: zod.number(),
  newLeadsToday: zod.number(),
  newLeadsYesterday: zod
    .number()
    .describe("New leads created yesterday (for today-vs-yesterday delta)."),
  newLeads: zod.number(),
  pendingReplies: zod.number(),
  assigned: zod.number(),
  converted: zod.number(),
  lost: zod.number(),
  activeProjects: zod.number(),
  followUpReminders: zod.number(),
  recentActivities: zod.array(
    zod.object({
      id: zod.number(),
      leadId: zod.number().nullish(),
      leadName: zod.string().nullish(),
      action: zod.string(),
      performedBy: zod.string().nullish(),
      createdAt: zod.coerce.date(),
    }),
  ),
  leadsByDay: zod
    .array(
      zod.object({
        date: zod.string().describe("ISO date (YYYY-MM-DD)"),
        count: zod.number(),
      }),
    )
    .describe("30-day series of new leads per day (oldest first)."),
  contentCounts: zod.object({
    projects: zod.number(),
    blogPosts: zod.number(),
    heroSlides: zod.number(),
    mediaArticles: zod.number(),
    instagramPosts: zod.number(),
    businessHighlights: zod.number(),
  }),
  currentPeriod: zod
    .object({
      totalLeads: zod.number(),
      newLeads: zod.number(),
      converted: zod.number(),
    })
    .describe("Counts for the current 7-day window, used for KPI deltas."),
  previousPeriod: zod
    .object({
      totalLeads: zod.number(),
      newLeads: zod.number(),
      converted: zod.number(),
    })
    .describe("Counts for the previous 7-day window, used for KPI deltas."),
});

/**
 * Paginated, filterable list of leads
 * @summary List leads
 */
export const listLeadsQueryPageDefault = 1;
export const listLeadsQueryLimitDefault = 20;
export const listLeadsQueryLimitMax = 100;

export const ListLeadsQueryParams = zod.object({
  page: zod.coerce.number().default(listLeadsQueryPageDefault),
  limit: zod.coerce
    .number()
    .max(listLeadsQueryLimitMax)
    .default(listLeadsQueryLimitDefault),
  status: zod
    .enum(["new", "pending", "assigned", "converted", "lost"])
    .optional(),
  search: zod.coerce.string().optional(),
});

export const ListLeadsResponse = zod.object({
  leads: zod.array(
    zod.object({
      id: zod.number(),
      name: zod.string(),
      email: zod.string().nullish(),
      phone: zod.string().nullish(),
      source: zod.string().nullish(),
      status: zod.enum(["new", "pending", "assigned", "converted", "lost"]),
      assignedTo: zod.string().nullish(),
      notes: zod.string().nullish(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
  total: zod.number(),
  page: zod.number(),
  limit: zod.number(),
});

/**
 * @summary Create a lead
 */
export const CreateLeadBody = zod.object({
  name: zod.string(),
  email: zod.string().optional(),
  phone: zod.string().optional(),
  source: zod.string().optional(),
  status: zod
    .enum(["new", "pending", "assigned", "converted", "lost"])
    .optional(),
  assignedTo: zod.string().optional(),
  notes: zod.string().optional(),
});

/**
 * @summary Update a lead
 */
export const UpdateLeadParams = zod.object({
  id: zod.coerce.number(),
});

export const UpdateLeadBody = zod.object({
  name: zod.string().optional(),
  email: zod.string().optional(),
  phone: zod.string().optional(),
  source: zod.string().optional(),
  status: zod
    .enum(["new", "pending", "assigned", "converted", "lost"])
    .optional(),
  assignedTo: zod.string().optional(),
  notes: zod.string().optional(),
});

export const UpdateLeadResponse = zod.object({
  id: zod.number(),
  name: zod.string(),
  email: zod.string().nullish(),
  phone: zod.string().nullish(),
  source: zod.string().nullish(),
  status: zod.enum(["new", "pending", "assigned", "converted", "lost"]),
  assignedTo: zod.string().nullish(),
  notes: zod.string().nullish(),
  createdAt: zod.coerce.date(),
  updatedAt: zod.coerce.date(),
});

/**
 * @summary Delete a lead
 */
export const DeleteLeadParams = zod.object({
  id: zod.coerce.number(),
});

export const DeleteLeadResponse = zod.object({
  success: zod.boolean().optional(),
});

/**
 * Returns the 20 most recent activity log entries
 * @summary Recent activities
 */
export const ListActivitiesResponse = zod.object({
  activities: zod.array(
    zod.object({
      id: zod.number(),
      leadId: zod.number().nullish(),
      leadName: zod.string().nullish(),
      action: zod.string(),
      performedBy: zod.string().nullish(),
      createdAt: zod.coerce.date(),
    }),
  ),
});

/**
 * Returns upcoming, unfinished follow-up reminders
 * @summary Upcoming reminders
 */
export const ListRemindersResponse = zod.object({
  reminders: zod.array(
    zod.object({
      id: zod.number(),
      leadId: zod.number().nullish(),
      leadName: zod.string().nullish(),
      dueAt: zod.coerce.date(),
      note: zod.string().nullish(),
      done: zod.boolean(),
      createdAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Mark reminder as done
 */
export const MarkReminderDoneParams = zod.object({
  id: zod.coerce.number(),
});

export const MarkReminderDoneResponse = zod.object({
  id: zod.number(),
  leadId: zod.number().nullish(),
  leadName: zod.string().nullish(),
  dueAt: zod.coerce.date(),
  note: zod.string().nullish(),
  done: zod.boolean(),
  createdAt: zod.coerce.date(),
});

/**
 * @summary List active projects
 */
export const ListProjectsQueryParams = zod.object({
  status: zod.enum(["ongoing", "completed", "upcoming"]).optional(),
  category: zod.enum(["RESIDENTIAL", "COMMERCIAL", "TOWNSHIP"]).optional(),
});

export const ListProjectsResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      slug: zod.string(),
      name: zod.string(),
      category: zod.enum(["RESIDENTIAL", "COMMERCIAL", "TOWNSHIP"]),
      status: zod.enum(["ongoing", "completed", "upcoming"]),
      address: zod.string().nullish(),
      location: zod.string().nullish(),
      priceRange: zod.string().nullish(),
      aboutDescription: zod.string().nullish(),
      reraNumber: zod.string().nullish(),
      googleMapsLink: zod.string().nullish(),
      heroImage1: zod.string().nullish(),
      heroImage2: zod.string().nullish(),
      heroImage3: zod.string().nullish(),
      logoImage: zod.string().nullish(),
      masterPlanImage: zod.string().nullish(),
      floorPlanImage: zod.string().nullish(),
      exclusiveClubImage: zod.string().nullish(),
      facilitiesNearbyImage: zod.string().nullish(),
      constructionUpdateImage: zod.string().nullish(),
      locationImage: zod.string().nullish(),
      brochureFile: zod.string().nullish(),
      videoUrl: zod.string().nullish(),
      reelUrl: zod.string().nullish(),
      heroImageUrl: zod.string().nullish(),
      isFeatured: zod.boolean().optional(),
      featuredOrder: zod.number().nullish(),
      featuredTabLabel: zod.string().nullish(),
      featuredTitle: zod.string().nullish(),
      featuredDescription: zod.string().nullish(),
      bedrooms: zod.string().nullish(),
      area: zod.string().nullish(),
      price: zod.string().nullish(),
      amenities: zod.array(zod.string()).nullish(),
      galleryImages: zod.array(zod.string()).nullish(),
      highlights: zod.array(zod.string()).optional(),
      faqs: zod
        .array(
          zod.object({
            question: zod.string(),
            answer: zod.string(),
          }),
        )
        .optional(),
      areaAndLot: zod.array(zod.string()).optional(),
      interiorFeatures: zod.array(zod.string()).optional(),
      exteriorFeatures: zod.array(zod.string()).optional(),
      isActive: zod.boolean(),
      sortOrder: zod.number(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Get a project by slug
 */
export const GetProjectParams = zod.object({
  slug: zod.coerce.string(),
});

export const GetProjectResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    slug: zod.string(),
    name: zod.string(),
    category: zod.enum(["RESIDENTIAL", "COMMERCIAL", "TOWNSHIP"]),
    status: zod.enum(["ongoing", "completed", "upcoming"]),
    address: zod.string().nullish(),
    location: zod.string().nullish(),
    priceRange: zod.string().nullish(),
    aboutDescription: zod.string().nullish(),
    reraNumber: zod.string().nullish(),
    googleMapsLink: zod.string().nullish(),
    heroImage1: zod.string().nullish(),
    heroImage2: zod.string().nullish(),
    heroImage3: zod.string().nullish(),
    logoImage: zod.string().nullish(),
    masterPlanImage: zod.string().nullish(),
    floorPlanImage: zod.string().nullish(),
    exclusiveClubImage: zod.string().nullish(),
    facilitiesNearbyImage: zod.string().nullish(),
    constructionUpdateImage: zod.string().nullish(),
    locationImage: zod.string().nullish(),
    brochureFile: zod.string().nullish(),
    videoUrl: zod.string().nullish(),
    reelUrl: zod.string().nullish(),
    heroImageUrl: zod.string().nullish(),
    isFeatured: zod.boolean().optional(),
    featuredOrder: zod.number().nullish(),
    featuredTabLabel: zod.string().nullish(),
    featuredTitle: zod.string().nullish(),
    featuredDescription: zod.string().nullish(),
    bedrooms: zod.string().nullish(),
    area: zod.string().nullish(),
    price: zod.string().nullish(),
    amenities: zod.array(zod.string()).nullish(),
    galleryImages: zod.array(zod.string()).nullish(),
    highlights: zod.array(zod.string()).optional(),
    faqs: zod
      .array(
        zod.object({
          question: zod.string(),
          answer: zod.string(),
        }),
      )
      .optional(),
    areaAndLot: zod.array(zod.string()).optional(),
    interiorFeatures: zod.array(zod.string()).optional(),
    exteriorFeatures: zod.array(zod.string()).optional(),
    isActive: zod.boolean(),
    sortOrder: zod.number(),
    createdAt: zod.coerce.date(),
    updatedAt: zod.coerce.date(),
  }),
});

/**
 * @summary List published blog posts
 */
export const ListBlogPostsQueryParams = zod.object({
  category: zod.coerce.string().optional(),
  limit: zod.coerce.number().optional(),
});

export const ListBlogPostsResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      slug: zod.string(),
      title: zod.string(),
      category: zod.string(),
      excerpt: zod.string().nullish(),
      content: zod.string().nullish(),
      heroImage: zod.string().nullish(),
      author: zod.string().nullish(),
      publishedAt: zod.coerce.date().nullish(),
      isPublished: zod.boolean(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Get a blog post by slug
 */
export const GetBlogPostParams = zod.object({
  slug: zod.coerce.string(),
});

export const GetBlogPostResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    slug: zod.string(),
    title: zod.string(),
    category: zod.string(),
    excerpt: zod.string().nullish(),
    content: zod.string().nullish(),
    heroImage: zod.string().nullish(),
    author: zod.string().nullish(),
    publishedAt: zod.coerce.date().nullish(),
    isPublished: zod.boolean(),
    createdAt: zod.coerce.date(),
    updatedAt: zod.coerce.date(),
  }),
});

/**
 * @summary List active job openings
 */
export const ListJobsResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      title: zod.string(),
      department: zod.string(),
      experience: zod.string(),
      description: zod.string().nullish(),
      isActive: zod.boolean(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary List published press articles
 */
export const ListMediaArticlesResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      title: zod.string(),
      source: zod.string().nullish(),
      category: zod.string().nullish(),
      excerpt: zod.string().nullish(),
      publishedDate: zod.string().nullish(),
      imageUrl: zod.string().nullish(),
      articleUrl: zod.string().nullish(),
      isPublished: zod.boolean(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary List gallery images
 */
export const ListGalleryImagesResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      imageUrl: zod.string(),
      caption: zod.string().nullish(),
      sortOrder: zod.number(),
      createdAt: zod.coerce.date(),
    }),
  ),
});

/**
 * Returns stats, milestones, awards, team members, and CSR initiatives in a single response
 * @summary Get all site-wide settings
 */
export const GetSiteSettingsResponse = zod.object({
  data: zod.object({
    stats: zod.array(
      zod.object({
        id: zod.number(),
        label: zod.string(),
        value: zod.number(),
        suffix: zod.string(),
        sortOrder: zod.number(),
        updatedAt: zod.coerce.date(),
      }),
    ),
    milestones: zod.array(
      zod.object({
        id: zod.number(),
        year: zod.string(),
        title: zod.string(),
        description: zod.string().nullish(),
        imageUrl: zod.string().nullish(),
        sortOrder: zod.number(),
        createdAt: zod.coerce.date(),
        updatedAt: zod.coerce.date(),
      }),
    ),
    awards: zod.array(
      zod.object({
        id: zod.number(),
        year: zod.string(),
        title: zod.string(),
        organization: zod.string().nullish(),
        imageUrl: zod.string().nullish(),
        sortOrder: zod.number(),
        createdAt: zod.coerce.date(),
      }),
    ),
    team: zod.array(
      zod.object({
        id: zod.number(),
        name: zod.string(),
        role: zod.string(),
        imageUrl: zod.string().nullish(),
        bio: zod.string().nullish(),
        sortOrder: zod.number(),
        section: zod.enum(["leadership", "management"]),
        createdAt: zod.coerce.date(),
        updatedAt: zod.coerce.date(),
      }),
    ),
    csr: zod.array(
      zod.object({
        id: zod.number(),
        title: zod.string(),
        description: zod.string().nullish(),
        imageUrl: zod.string().nullish(),
        sortOrder: zod.number(),
        createdAt: zod.coerce.date(),
        updatedAt: zod.coerce.date(),
      }),
    ),
  }),
});

/**
 * @summary List all projects (admin)
 */
export const AdminListProjectsResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      slug: zod.string(),
      name: zod.string(),
      category: zod.enum(["RESIDENTIAL", "COMMERCIAL", "TOWNSHIP"]),
      status: zod.enum(["ongoing", "completed", "upcoming"]),
      address: zod.string().nullish(),
      location: zod.string().nullish(),
      priceRange: zod.string().nullish(),
      aboutDescription: zod.string().nullish(),
      reraNumber: zod.string().nullish(),
      googleMapsLink: zod.string().nullish(),
      heroImage1: zod.string().nullish(),
      heroImage2: zod.string().nullish(),
      heroImage3: zod.string().nullish(),
      logoImage: zod.string().nullish(),
      masterPlanImage: zod.string().nullish(),
      floorPlanImage: zod.string().nullish(),
      exclusiveClubImage: zod.string().nullish(),
      facilitiesNearbyImage: zod.string().nullish(),
      constructionUpdateImage: zod.string().nullish(),
      locationImage: zod.string().nullish(),
      brochureFile: zod.string().nullish(),
      videoUrl: zod.string().nullish(),
      reelUrl: zod.string().nullish(),
      heroImageUrl: zod.string().nullish(),
      isFeatured: zod.boolean().optional(),
      featuredOrder: zod.number().nullish(),
      featuredTabLabel: zod.string().nullish(),
      featuredTitle: zod.string().nullish(),
      featuredDescription: zod.string().nullish(),
      bedrooms: zod.string().nullish(),
      area: zod.string().nullish(),
      price: zod.string().nullish(),
      amenities: zod.array(zod.string()).nullish(),
      galleryImages: zod.array(zod.string()).nullish(),
      highlights: zod.array(zod.string()).optional(),
      faqs: zod
        .array(
          zod.object({
            question: zod.string(),
            answer: zod.string(),
          }),
        )
        .optional(),
      areaAndLot: zod.array(zod.string()).optional(),
      interiorFeatures: zod.array(zod.string()).optional(),
      exteriorFeatures: zod.array(zod.string()).optional(),
      isActive: zod.boolean(),
      sortOrder: zod.number(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Create a project
 */
export const AdminCreateProjectBody = zod.object({
  slug: zod.string(),
  name: zod.string(),
  category: zod.enum(["RESIDENTIAL", "COMMERCIAL", "TOWNSHIP"]),
  status: zod.enum(["ongoing", "completed", "upcoming"]).optional(),
  address: zod.string().optional(),
  location: zod.string().optional(),
  priceRange: zod.string().optional(),
  aboutDescription: zod.string().optional(),
  reraNumber: zod.string().optional(),
  googleMapsLink: zod.string().optional(),
  heroImage1: zod.string().optional(),
  heroImage2: zod.string().optional(),
  heroImage3: zod.string().optional(),
  logoImage: zod.string().optional(),
  masterPlanImage: zod.string().optional(),
  floorPlanImage: zod.string().optional(),
  exclusiveClubImage: zod.string().optional(),
  facilitiesNearbyImage: zod.string().optional(),
  constructionUpdateImage: zod.string().optional(),
  locationImage: zod.string().optional(),
  brochureFile: zod.string().optional(),
  videoUrl: zod.string().optional(),
  highlights: zod.array(zod.string()).optional(),
  faqs: zod.array(zod.object({}).passthrough()).optional(),
  areaAndLot: zod.array(zod.string()).optional(),
  interiorFeatures: zod.array(zod.string()).optional(),
  exteriorFeatures: zod.array(zod.string()).optional(),
  isActive: zod.boolean().optional(),
  sortOrder: zod.number().optional(),
});

/**
 * @summary Update a project
 */
export const AdminUpdateProjectParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateProjectBody = zod.object({
  slug: zod.string(),
  name: zod.string(),
  category: zod.enum(["RESIDENTIAL", "COMMERCIAL", "TOWNSHIP"]),
  status: zod.enum(["ongoing", "completed", "upcoming"]).optional(),
  address: zod.string().optional(),
  location: zod.string().optional(),
  priceRange: zod.string().optional(),
  aboutDescription: zod.string().optional(),
  reraNumber: zod.string().optional(),
  googleMapsLink: zod.string().optional(),
  heroImage1: zod.string().optional(),
  heroImage2: zod.string().optional(),
  heroImage3: zod.string().optional(),
  logoImage: zod.string().optional(),
  masterPlanImage: zod.string().optional(),
  floorPlanImage: zod.string().optional(),
  exclusiveClubImage: zod.string().optional(),
  facilitiesNearbyImage: zod.string().optional(),
  constructionUpdateImage: zod.string().optional(),
  locationImage: zod.string().optional(),
  brochureFile: zod.string().optional(),
  videoUrl: zod.string().optional(),
  highlights: zod.array(zod.string()).optional(),
  faqs: zod.array(zod.object({}).passthrough()).optional(),
  areaAndLot: zod.array(zod.string()).optional(),
  interiorFeatures: zod.array(zod.string()).optional(),
  exteriorFeatures: zod.array(zod.string()).optional(),
  isActive: zod.boolean().optional(),
  sortOrder: zod.number().optional(),
});

export const AdminUpdateProjectResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    slug: zod.string(),
    name: zod.string(),
    category: zod.enum(["RESIDENTIAL", "COMMERCIAL", "TOWNSHIP"]),
    status: zod.enum(["ongoing", "completed", "upcoming"]),
    address: zod.string().nullish(),
    location: zod.string().nullish(),
    priceRange: zod.string().nullish(),
    aboutDescription: zod.string().nullish(),
    reraNumber: zod.string().nullish(),
    googleMapsLink: zod.string().nullish(),
    heroImage1: zod.string().nullish(),
    heroImage2: zod.string().nullish(),
    heroImage3: zod.string().nullish(),
    logoImage: zod.string().nullish(),
    masterPlanImage: zod.string().nullish(),
    floorPlanImage: zod.string().nullish(),
    exclusiveClubImage: zod.string().nullish(),
    facilitiesNearbyImage: zod.string().nullish(),
    constructionUpdateImage: zod.string().nullish(),
    locationImage: zod.string().nullish(),
    brochureFile: zod.string().nullish(),
    videoUrl: zod.string().nullish(),
    reelUrl: zod.string().nullish(),
    heroImageUrl: zod.string().nullish(),
    isFeatured: zod.boolean().optional(),
    featuredOrder: zod.number().nullish(),
    featuredTabLabel: zod.string().nullish(),
    featuredTitle: zod.string().nullish(),
    featuredDescription: zod.string().nullish(),
    bedrooms: zod.string().nullish(),
    area: zod.string().nullish(),
    price: zod.string().nullish(),
    amenities: zod.array(zod.string()).nullish(),
    galleryImages: zod.array(zod.string()).nullish(),
    highlights: zod.array(zod.string()).optional(),
    faqs: zod
      .array(
        zod.object({
          question: zod.string(),
          answer: zod.string(),
        }),
      )
      .optional(),
    areaAndLot: zod.array(zod.string()).optional(),
    interiorFeatures: zod.array(zod.string()).optional(),
    exteriorFeatures: zod.array(zod.string()).optional(),
    isActive: zod.boolean(),
    sortOrder: zod.number(),
    createdAt: zod.coerce.date(),
    updatedAt: zod.coerce.date(),
  }),
});

/**
 * @summary Delete a project
 */
export const AdminDeleteProjectParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List all blog posts (admin)
 */
export const AdminListBlogPostsResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      slug: zod.string(),
      title: zod.string(),
      category: zod.string(),
      excerpt: zod.string().nullish(),
      content: zod.string().nullish(),
      heroImage: zod.string().nullish(),
      author: zod.string().nullish(),
      publishedAt: zod.coerce.date().nullish(),
      isPublished: zod.boolean(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Create a blog post
 */
export const AdminCreateBlogPostBody = zod.object({
  slug: zod.string(),
  title: zod.string(),
  category: zod.string(),
  excerpt: zod.string().optional(),
  content: zod.string().optional(),
  heroImage: zod.string().optional(),
  author: zod.string().optional(),
  publishedAt: zod.coerce.date().optional(),
  isPublished: zod.boolean().optional(),
});

/**
 * @summary Update a blog post
 */
export const AdminUpdateBlogPostParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateBlogPostBody = zod.object({
  slug: zod.string(),
  title: zod.string(),
  category: zod.string(),
  excerpt: zod.string().optional(),
  content: zod.string().optional(),
  heroImage: zod.string().optional(),
  author: zod.string().optional(),
  publishedAt: zod.coerce.date().optional(),
  isPublished: zod.boolean().optional(),
});

export const AdminUpdateBlogPostResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    slug: zod.string(),
    title: zod.string(),
    category: zod.string(),
    excerpt: zod.string().nullish(),
    content: zod.string().nullish(),
    heroImage: zod.string().nullish(),
    author: zod.string().nullish(),
    publishedAt: zod.coerce.date().nullish(),
    isPublished: zod.boolean(),
    createdAt: zod.coerce.date(),
    updatedAt: zod.coerce.date(),
  }),
});

/**
 * @summary Delete a blog post
 */
export const AdminDeleteBlogPostParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List all job openings (admin)
 */
export const AdminListJobsResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      title: zod.string(),
      department: zod.string(),
      experience: zod.string(),
      description: zod.string().nullish(),
      isActive: zod.boolean(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Create a job opening
 */
export const AdminCreateJobBody = zod.object({
  title: zod.string(),
  department: zod.string(),
  experience: zod.string(),
  description: zod.string().optional(),
  isActive: zod.boolean().optional(),
});

/**
 * @summary Update a job opening
 */
export const AdminUpdateJobParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateJobBody = zod.object({
  title: zod.string(),
  department: zod.string(),
  experience: zod.string(),
  description: zod.string().optional(),
  isActive: zod.boolean().optional(),
});

export const AdminUpdateJobResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    title: zod.string(),
    department: zod.string(),
    experience: zod.string(),
    description: zod.string().nullish(),
    isActive: zod.boolean(),
    createdAt: zod.coerce.date(),
    updatedAt: zod.coerce.date(),
  }),
});

/**
 * @summary Delete a job opening
 */
export const AdminDeleteJobParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List all media articles (admin)
 */
export const AdminListMediaArticlesResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      title: zod.string(),
      source: zod.string().nullish(),
      category: zod.string().nullish(),
      excerpt: zod.string().nullish(),
      publishedDate: zod.string().nullish(),
      imageUrl: zod.string().nullish(),
      articleUrl: zod.string().nullish(),
      isPublished: zod.boolean(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Create a media article
 */
export const AdminCreateMediaArticleBody = zod.object({
  title: zod.string(),
  source: zod.string().optional(),
  category: zod.string().optional(),
  excerpt: zod.string().optional(),
  publishedDate: zod.string().optional(),
  imageUrl: zod.string().optional(),
  articleUrl: zod.string().optional(),
  isPublished: zod.boolean().optional(),
});

/**
 * @summary Update a media article
 */
export const AdminUpdateMediaArticleParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateMediaArticleBody = zod.object({
  title: zod.string(),
  source: zod.string().optional(),
  category: zod.string().optional(),
  excerpt: zod.string().optional(),
  publishedDate: zod.string().optional(),
  imageUrl: zod.string().optional(),
  articleUrl: zod.string().optional(),
  isPublished: zod.boolean().optional(),
});

export const AdminUpdateMediaArticleResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    title: zod.string(),
    source: zod.string().nullish(),
    category: zod.string().nullish(),
    excerpt: zod.string().nullish(),
    publishedDate: zod.string().nullish(),
    imageUrl: zod.string().nullish(),
    articleUrl: zod.string().nullish(),
    isPublished: zod.boolean(),
    createdAt: zod.coerce.date(),
    updatedAt: zod.coerce.date(),
  }),
});

/**
 * @summary Delete a media article
 */
export const AdminDeleteMediaArticleParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List all gallery images (admin)
 */
export const AdminListGalleryImagesResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      imageUrl: zod.string(),
      caption: zod.string().nullish(),
      sortOrder: zod.number(),
      createdAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Add a gallery image
 */
export const AdminCreateGalleryImageBody = zod.object({
  imageUrl: zod.string(),
  caption: zod.string().optional(),
  sortOrder: zod.number().optional(),
});

/**
 * @summary Update a gallery image
 */
export const AdminUpdateGalleryImageParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateGalleryImageBody = zod.object({
  imageUrl: zod.string(),
  caption: zod.string().optional(),
  sortOrder: zod.number().optional(),
});

export const AdminUpdateGalleryImageResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    imageUrl: zod.string(),
    caption: zod.string().nullish(),
    sortOrder: zod.number(),
    createdAt: zod.coerce.date(),
  }),
});

/**
 * @summary Delete a gallery image
 */
export const AdminDeleteGalleryImageParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List site stats
 */
export const AdminListSiteStatsResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      label: zod.string(),
      value: zod.number(),
      suffix: zod.string(),
      sortOrder: zod.number(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Create a site stat
 */
export const AdminCreateSiteStatBody = zod.object({
  label: zod.string(),
  value: zod.number(),
  suffix: zod.string().optional(),
  sortOrder: zod.number().optional(),
});

/**
 * @summary Update a site stat
 */
export const AdminUpdateSiteStatParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateSiteStatBody = zod.object({
  label: zod.string(),
  value: zod.number(),
  suffix: zod.string().optional(),
  sortOrder: zod.number().optional(),
});

export const AdminUpdateSiteStatResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    label: zod.string(),
    value: zod.number(),
    suffix: zod.string(),
    sortOrder: zod.number(),
    updatedAt: zod.coerce.date(),
  }),
});

/**
 * @summary Delete a site stat
 */
export const AdminDeleteSiteStatParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List milestones
 */
export const AdminListMilestonesResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      year: zod.string(),
      title: zod.string(),
      description: zod.string().nullish(),
      imageUrl: zod.string().nullish(),
      sortOrder: zod.number(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Create a milestone
 */
export const AdminCreateMilestoneBody = zod.object({
  year: zod.string(),
  title: zod.string(),
  description: zod.string().optional(),
  imageUrl: zod.string().optional(),
  sortOrder: zod.number().optional(),
});

/**
 * @summary Update a milestone
 */
export const AdminUpdateMilestoneParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateMilestoneBody = zod.object({
  year: zod.string(),
  title: zod.string(),
  description: zod.string().optional(),
  imageUrl: zod.string().optional(),
  sortOrder: zod.number().optional(),
});

export const AdminUpdateMilestoneResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    year: zod.string(),
    title: zod.string(),
    description: zod.string().nullish(),
    imageUrl: zod.string().nullish(),
    sortOrder: zod.number(),
    createdAt: zod.coerce.date(),
    updatedAt: zod.coerce.date(),
  }),
});

/**
 * @summary Delete a milestone
 */
export const AdminDeleteMilestoneParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List awards
 */
export const AdminListAwardsResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      year: zod.string(),
      title: zod.string(),
      organization: zod.string().nullish(),
      imageUrl: zod.string().nullish(),
      sortOrder: zod.number(),
      createdAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Create an award
 */
export const AdminCreateAwardBody = zod.object({
  year: zod.string(),
  title: zod.string(),
  organization: zod.string().optional(),
  imageUrl: zod.string().optional(),
  sortOrder: zod.number().optional(),
});

/**
 * @summary Update an award
 */
export const AdminUpdateAwardParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateAwardBody = zod.object({
  year: zod.string(),
  title: zod.string(),
  organization: zod.string().optional(),
  imageUrl: zod.string().optional(),
  sortOrder: zod.number().optional(),
});

export const AdminUpdateAwardResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    year: zod.string(),
    title: zod.string(),
    organization: zod.string().nullish(),
    imageUrl: zod.string().nullish(),
    sortOrder: zod.number(),
    createdAt: zod.coerce.date(),
  }),
});

/**
 * @summary Delete an award
 */
export const AdminDeleteAwardParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List team members
 */
export const AdminListTeamMembersResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      name: zod.string(),
      role: zod.string(),
      imageUrl: zod.string().nullish(),
      bio: zod.string().nullish(),
      sortOrder: zod.number(),
      section: zod.enum(["leadership", "management"]),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Create a team member
 */
export const AdminCreateTeamMemberBody = zod.object({
  name: zod.string(),
  role: zod.string(),
  imageUrl: zod.string().optional(),
  bio: zod.string().optional(),
  sortOrder: zod.number().optional(),
  section: zod.enum(["leadership", "management"]),
});

/**
 * @summary Update a team member
 */
export const AdminUpdateTeamMemberParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateTeamMemberBody = zod.object({
  name: zod.string(),
  role: zod.string(),
  imageUrl: zod.string().optional(),
  bio: zod.string().optional(),
  sortOrder: zod.number().optional(),
  section: zod.enum(["leadership", "management"]),
});

export const AdminUpdateTeamMemberResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    name: zod.string(),
    role: zod.string(),
    imageUrl: zod.string().nullish(),
    bio: zod.string().nullish(),
    sortOrder: zod.number(),
    section: zod.enum(["leadership", "management"]),
    createdAt: zod.coerce.date(),
    updatedAt: zod.coerce.date(),
  }),
});

/**
 * @summary Delete a team member
 */
export const AdminDeleteTeamMemberParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List CSR initiatives
 */
export const AdminListCsrInitiativesResponse = zod.object({
  data: zod.array(
    zod.object({
      id: zod.number(),
      title: zod.string(),
      description: zod.string().nullish(),
      imageUrl: zod.string().nullish(),
      sortOrder: zod.number(),
      createdAt: zod.coerce.date(),
      updatedAt: zod.coerce.date(),
    }),
  ),
});

/**
 * @summary Create a CSR initiative
 */
export const AdminCreateCsrInitiativeBody = zod.object({
  title: zod.string(),
  description: zod.string().optional(),
  imageUrl: zod.string().optional(),
  sortOrder: zod.number().optional(),
});

/**
 * @summary Update a CSR initiative
 */
export const AdminUpdateCsrInitiativeParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateCsrInitiativeBody = zod.object({
  title: zod.string(),
  description: zod.string().optional(),
  imageUrl: zod.string().optional(),
  sortOrder: zod.number().optional(),
});

export const AdminUpdateCsrInitiativeResponse = zod.object({
  data: zod.object({
    id: zod.number(),
    title: zod.string(),
    description: zod.string().nullish(),
    imageUrl: zod.string().nullish(),
    sortOrder: zod.number(),
    createdAt: zod.coerce.date(),
    updatedAt: zod.coerce.date(),
  }),
});

/**
 * @summary Delete a CSR initiative
 */
export const AdminDeleteCsrInitiativeParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List active hero slides
 */
export const ListHeroSlidesResponse = zod.object({
  data: zod
    .array(
      zod.object({
        id: zod.number(),
        title: zod.string(),
        subtitle: zod.string().nullish(),
        description: zod.string().nullish(),
        backgroundVideoUrl: zod.string().nullish(),
        backgroundImageUrl: zod.string().nullish(),
        ctaLabel: zod.string().nullish(),
        ctaLink: zod.string().nullish(),
        sortOrder: zod.number(),
        isActive: zod.boolean(),
        createdAt: zod.coerce.date().optional(),
        updatedAt: zod.coerce.date().optional(),
      }),
    )
    .optional(),
});

/**
 * @summary List active business highlights
 */
export const ListBusinessHighlightsResponse = zod.object({
  data: zod
    .array(
      zod.object({
        id: zod.number(),
        projectName: zod.string(),
        slug: zod.string().nullish(),
        description: zod.string().nullish(),
        imageUrl: zod.string().nullish(),
        sortOrder: zod.number(),
        isActive: zod.boolean(),
        createdAt: zod.coerce.date().optional(),
        updatedAt: zod.coerce.date().optional(),
      }),
    )
    .optional(),
});

/**
 * @summary List active client logos
 */
export const ListClientLogosResponse = zod.object({
  data: zod
    .array(
      zod.object({
        id: zod.number(),
        name: zod.string().nullish(),
        imageUrl: zod.string().nullish(),
        website: zod.string().nullish(),
        sortOrder: zod.number(),
        isActive: zod.boolean(),
        createdAt: zod.coerce.date().optional(),
        updatedAt: zod.coerce.date().optional(),
      }),
    )
    .optional(),
});

/**
 * @summary Get all site config as key/value map
 */
export const GetSiteConfigResponse = zod.object({
  data: zod.record(zod.string(), zod.string().nullable()).optional(),
});

/**
 * @summary List career page content grouped by section
 */
export const ListCareerContentQueryParams = zod.object({
  section: zod.enum(["life_at_ub", "why_join", "expectation"]).optional(),
});

export const ListCareerContentResponse = zod.object({
  data: zod
    .union([
      zod.record(
        zod.string(),
        zod.array(
          zod.object({
            id: zod.number(),
            section: zod.enum(["life_at_ub", "why_join", "expectation"]),
            title: zod.string(),
            description: zod.string().nullish(),
            imageUrl: zod.string().nullish(),
            icon: zod.string().nullish(),
            sortOrder: zod.number(),
            isActive: zod.boolean(),
            createdAt: zod.coerce.date().optional(),
            updatedAt: zod.coerce.date().optional(),
          }),
        ),
      ),
      zod.array(
        zod.object({
          id: zod.number(),
          section: zod.enum(["life_at_ub", "why_join", "expectation"]),
          title: zod.string(),
          description: zod.string().nullish(),
          imageUrl: zod.string().nullish(),
          icon: zod.string().nullish(),
          sortOrder: zod.number(),
          isActive: zod.boolean(),
          createdAt: zod.coerce.date().optional(),
          updatedAt: zod.coerce.date().optional(),
        }),
      ),
    ])
    .optional(),
});

/**
 * @summary List active Instagram posts
 */
export const ListInstagramPostsResponse = zod.object({
  data: zod
    .array(
      zod.object({
        id: zod.number(),
        postId: zod.string().nullish(),
        imageUrl: zod.string().nullish(),
        link: zod.string(),
        altText: zod.string().nullish(),
        sortOrder: zod.number(),
        isActive: zod.boolean(),
        createdAt: zod.coerce.date().optional(),
        updatedAt: zod.coerce.date().optional(),
      }),
    )
    .optional(),
});

/**
 * @summary List all hero slides (admin)
 */
export const AdminListHeroSlidesResponse = zod.object({
  data: zod
    .array(
      zod.object({
        id: zod.number(),
        title: zod.string(),
        subtitle: zod.string().nullish(),
        description: zod.string().nullish(),
        backgroundVideoUrl: zod.string().nullish(),
        backgroundImageUrl: zod.string().nullish(),
        ctaLabel: zod.string().nullish(),
        ctaLink: zod.string().nullish(),
        sortOrder: zod.number(),
        isActive: zod.boolean(),
        createdAt: zod.coerce.date().optional(),
        updatedAt: zod.coerce.date().optional(),
      }),
    )
    .optional(),
});

/**
 * @summary Create a hero slide
 */
export const AdminCreateHeroSlideBody = zod.object({
  id: zod.number(),
  title: zod.string(),
  subtitle: zod.string().nullish(),
  description: zod.string().nullish(),
  backgroundVideoUrl: zod.string().nullish(),
  backgroundImageUrl: zod.string().nullish(),
  ctaLabel: zod.string().nullish(),
  ctaLink: zod.string().nullish(),
  sortOrder: zod.number(),
  isActive: zod.boolean(),
  createdAt: zod.coerce.date().optional(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Update a hero slide
 */
export const AdminUpdateHeroSlideParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateHeroSlideBody = zod.object({
  id: zod.number(),
  title: zod.string(),
  subtitle: zod.string().nullish(),
  description: zod.string().nullish(),
  backgroundVideoUrl: zod.string().nullish(),
  backgroundImageUrl: zod.string().nullish(),
  ctaLabel: zod.string().nullish(),
  ctaLink: zod.string().nullish(),
  sortOrder: zod.number(),
  isActive: zod.boolean(),
  createdAt: zod.coerce.date().optional(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Delete a hero slide
 */
export const AdminDeleteHeroSlideParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List all business highlights (admin)
 */
export const AdminListBusinessHighlightsResponse = zod.object({
  data: zod
    .array(
      zod.object({
        id: zod.number(),
        projectName: zod.string(),
        slug: zod.string().nullish(),
        description: zod.string().nullish(),
        imageUrl: zod.string().nullish(),
        sortOrder: zod.number(),
        isActive: zod.boolean(),
        createdAt: zod.coerce.date().optional(),
        updatedAt: zod.coerce.date().optional(),
      }),
    )
    .optional(),
});

/**
 * @summary Create a business highlight
 */
export const AdminCreateBusinessHighlightBody = zod.object({
  id: zod.number(),
  projectName: zod.string(),
  slug: zod.string().nullish(),
  description: zod.string().nullish(),
  imageUrl: zod.string().nullish(),
  sortOrder: zod.number(),
  isActive: zod.boolean(),
  createdAt: zod.coerce.date().optional(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Update a business highlight
 */
export const AdminUpdateBusinessHighlightParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateBusinessHighlightBody = zod.object({
  id: zod.number(),
  projectName: zod.string(),
  slug: zod.string().nullish(),
  description: zod.string().nullish(),
  imageUrl: zod.string().nullish(),
  sortOrder: zod.number(),
  isActive: zod.boolean(),
  createdAt: zod.coerce.date().optional(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Delete a business highlight
 */
export const AdminDeleteBusinessHighlightParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List all client logos (admin)
 */
export const AdminListClientLogosResponse = zod.object({
  data: zod
    .array(
      zod.object({
        id: zod.number(),
        name: zod.string().nullish(),
        imageUrl: zod.string().nullish(),
        website: zod.string().nullish(),
        sortOrder: zod.number(),
        isActive: zod.boolean(),
        createdAt: zod.coerce.date().optional(),
        updatedAt: zod.coerce.date().optional(),
      }),
    )
    .optional(),
});

/**
 * @summary Create a client logo
 */
export const AdminCreateClientLogoBody = zod.object({
  id: zod.number(),
  name: zod.string().nullish(),
  imageUrl: zod.string().nullish(),
  website: zod.string().nullish(),
  sortOrder: zod.number(),
  isActive: zod.boolean(),
  createdAt: zod.coerce.date().optional(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Update a client logo
 */
export const AdminUpdateClientLogoParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateClientLogoBody = zod.object({
  id: zod.number(),
  name: zod.string().nullish(),
  imageUrl: zod.string().nullish(),
  website: zod.string().nullish(),
  sortOrder: zod.number(),
  isActive: zod.boolean(),
  createdAt: zod.coerce.date().optional(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Delete a client logo
 */
export const AdminDeleteClientLogoParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List all site config entries (admin)
 */
export const AdminListSiteConfigResponse = zod.object({
  data: zod
    .array(
      zod.object({
        id: zod.number(),
        key: zod.string(),
        value: zod.string().nullish(),
        label: zod.string().nullish(),
        updatedAt: zod.coerce.date().optional(),
      }),
    )
    .optional(),
});

/**
 * @summary Create a site config entry
 */
export const AdminCreateSiteConfigBody = zod.object({
  id: zod.number(),
  key: zod.string(),
  value: zod.string().nullish(),
  label: zod.string().nullish(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Update a site config entry by key
 */
export const AdminUpdateSiteConfigParams = zod.object({
  key: zod.coerce.string(),
});

export const AdminUpdateSiteConfigBody = zod.object({
  id: zod.number(),
  key: zod.string(),
  value: zod.string().nullish(),
  label: zod.string().nullish(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Delete a site config entry by key
 */
export const AdminDeleteSiteConfigParams = zod.object({
  key: zod.coerce.string(),
});

/**
 * @summary List all career content (admin)
 */
export const AdminListCareerContentQueryParams = zod.object({
  section: zod.enum(["life_at_ub", "why_join", "expectation"]).optional(),
});

export const AdminListCareerContentResponse = zod.object({
  data: zod
    .array(
      zod.object({
        id: zod.number(),
        section: zod.enum(["life_at_ub", "why_join", "expectation"]),
        title: zod.string(),
        description: zod.string().nullish(),
        imageUrl: zod.string().nullish(),
        icon: zod.string().nullish(),
        sortOrder: zod.number(),
        isActive: zod.boolean(),
        createdAt: zod.coerce.date().optional(),
        updatedAt: zod.coerce.date().optional(),
      }),
    )
    .optional(),
});

/**
 * @summary Create a career content entry
 */
export const AdminCreateCareerContentBody = zod.object({
  id: zod.number(),
  section: zod.enum(["life_at_ub", "why_join", "expectation"]),
  title: zod.string(),
  description: zod.string().nullish(),
  imageUrl: zod.string().nullish(),
  icon: zod.string().nullish(),
  sortOrder: zod.number(),
  isActive: zod.boolean(),
  createdAt: zod.coerce.date().optional(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Update a career content entry
 */
export const AdminUpdateCareerContentParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateCareerContentBody = zod.object({
  id: zod.number(),
  section: zod.enum(["life_at_ub", "why_join", "expectation"]),
  title: zod.string(),
  description: zod.string().nullish(),
  imageUrl: zod.string().nullish(),
  icon: zod.string().nullish(),
  sortOrder: zod.number(),
  isActive: zod.boolean(),
  createdAt: zod.coerce.date().optional(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Delete a career content entry
 */
export const AdminDeleteCareerContentParams = zod.object({
  id: zod.coerce.number(),
});

/**
 * @summary List all Instagram posts (admin)
 */
export const AdminListInstagramPostsResponse = zod.object({
  data: zod
    .array(
      zod.object({
        id: zod.number(),
        postId: zod.string().nullish(),
        imageUrl: zod.string().nullish(),
        link: zod.string(),
        altText: zod.string().nullish(),
        sortOrder: zod.number(),
        isActive: zod.boolean(),
        createdAt: zod.coerce.date().optional(),
        updatedAt: zod.coerce.date().optional(),
      }),
    )
    .optional(),
});

/**
 * @summary Create an Instagram post
 */
export const AdminCreateInstagramPostBody = zod.object({
  id: zod.number(),
  postId: zod.string().nullish(),
  imageUrl: zod.string().nullish(),
  link: zod.string(),
  altText: zod.string().nullish(),
  sortOrder: zod.number(),
  isActive: zod.boolean(),
  createdAt: zod.coerce.date().optional(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Update an Instagram post
 */
export const AdminUpdateInstagramPostParams = zod.object({
  id: zod.coerce.number(),
});

export const AdminUpdateInstagramPostBody = zod.object({
  id: zod.number(),
  postId: zod.string().nullish(),
  imageUrl: zod.string().nullish(),
  link: zod.string(),
  altText: zod.string().nullish(),
  sortOrder: zod.number(),
  isActive: zod.boolean(),
  createdAt: zod.coerce.date().optional(),
  updatedAt: zod.coerce.date().optional(),
});

/**
 * @summary Delete an Instagram post
 */
export const AdminDeleteInstagramPostParams = zod.object({
  id: zod.coerce.number(),
});
